analyses/survival-analysis/03-survival-tp53.Rmd

```diff
@@ -0,0 +1,248 @@
+---
+title: "tp53_survival_analysis"
+author: "Zhuangzhuang Geng"
+date: "2023-09-20"
+output: html_document
+---
+
+
+## load libraries
+
+```{r libraries}
+suppressWarnings({
+  library(tidyverse)
+  library(survival)
+  library(ggplot2)
+  library(patchwork)
+})
+```
+
+## set directories and source functions for plotting 
+
+```{r}
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+analyses_dir <- file.path(root_dir, "analyses", "survival-analysis")
+input_dir <- file.path(analyses_dir, "input")
+plots_dir <- file.path(analyses_dir, "plots")
+
+## source this script for plotting
+source(file.path(analyses_dir, "utils", "survival_models.R"))
+```
+
+### read in file and
+### transfomr OS_days to OS_years
+
+```{r}
+HGG_survival <- read_tsv(file.path(input_dir, "HGG_survial_tp53.tsv"))
+
+HGG_survival <- HGG_survival %>%
+  mutate(OS_years = (OS_days / 365.25), 
+         binned_tp53 = replace_na(binned_tp53, "no tp53 score"), 
+         plot_mol_subtype = case_when(grepl("IHG", molecular_subtype) ~ "IHG", 
+                                      TRUE ~ molecular_subtype))
+
+```
+
+## plot the Kaplan–Meier survival curve on 
+## broad_mol, mol_subtype_without_tp53, gender, binned age, race and binned_tp53 score
+
+```{r}
+var <- c("reported_gender", "binned_age", "binned_tp53", "broad_mol", "mol_subtype_without_tp53", "race")
+
+for(i in var){
+  
+  kap_OS <- survival_analysis(
+  metadata = HGG_survival, 
+  ind_var = i,
+  test = "kap.meier",
+  metadata_sample_col = "sample_id",
+  days_col = "OS_days",
+  status_col = "OS_status"
+  )
+print(i)
+  plot <- plotKM(model = kap_OS, 
+       variable = i, 
+       combined = FALSE, 
+       title = i)
```

Comment:
I got the following error when running this for `i == "broad_mol"`: 

```
Error in ggsurvplot_df(d, fun = fun, color = color, palette = palette,  : 
  The length of legend.labs should be 5
```

I think this is due to NA being a value for `broad_mol`, and the current `plotKM()` function not knowing how to handle that. I've made a suggestion in the `survival_models.R` script that would take care of this, but another option would be to change this from NA to some other value ("not subtyped", or something similar).